src/main/java/dbaccessor/message/MessageAccessor.java

```diff
@@ -34,6 +39,36 @@ public void updateReceivedTs(Message message) {
         spannerTemplate.update(message, "MessageID", "ReceivedTS");
     }
 
+    /**
+     * Completes all DB insertions for the CreateMessage API in a single transaction.
+     */
+    public boolean insertForCreateMessageTransaction(Message message, Chat chat) {
+        return spannerOperations.performReadWriteTransaction(
+            transactionSpannerOperations -> {
+                transactionSpannerOperations.insert(message);
+                transactionSpannerOperations.update(chat, "ChatID", "LastSentMessageID");
+
+                return true;
+            }
+        );
+    }
+
+    /**
+     * Completes all DB insertions for the ListMessages API in a single transaction.
+     */
+    public boolean updateReceivedTsForListMessagesTransaction(List<Message> messageList) {
```

Comment:
`updateReceivedTimestampForMessages`

Reply:
Can I rename it as `updateReceivedTsForMessages`? Since I've used `Ts` everywhere in the code.

Reply:
Done